src/libcore/cell.rs

```diff
@@ -1267,3 +1269,34 @@ fn assert_coerce_unsized(a: UnsafeCell<&i32>, b: Cell<&i32>, c: RefCell<&i32>) {
     let _: Cell<&Send> = b;
     let _: RefCell<&Send> = c;
 }
+
+/// A cell that is always movable, even if the interior type is immovable.
+/// This prevents pointers to the inner type which means it can never be observed.
+#[cfg(not(stage0))]
+#[unstable(feature = "immovable_types", issue = "0")]
+#[lang = "movable_cell"]
+#[derive(Eq, PartialEq, Ord, PartialOrd, Copy, Clone, Hash, Debug)]
```

Comment:
These shouldn't be derived as the implementations would take pointers to the inner value.